### ISSUE TITLE
API: `s` argument different in scipy.fft and numpy.fft

### ISSUE THREAD

**Issue by peterbell10** [scipy/scipy#10569]
The nd transforms in `scipy.fft` and `numpy.fft` both provide different ways to ignore a value in the shape parameter `s`.

```python
import numpy as np
from scipy import sp_fft

x = np.arange(100).reshape(10,10)
# This works
assert np.fft.fftn(x, s=(5,None)).shape == (5, 10)
assert sp_fft.fftn(x, s=(5,-1)).shape   == (5, 10)
# But these don't
sp_fft.fftn(x, s=(5,None)).shape
np.fft.fftn(x, s=(5,-1)).shape
```

As far as I can tell, numpy's behaviour is not explicitly documented but I discovered this in `cupy`'s test suite which would suggest that people are using it.

I plan to add this but won't open a PR yet as it would conflict with #10518.

**Comment by grlee77** [comment]
By "plan to add this" do you mean allowing `None` in addition to `-1` for SciPy?  I prefer `-1`, because that is consistent with the usage in `numpy.reshape`. I am fine with allowing both for better interoperability, though.

**Comment by peterbell10** [comment]
>By "plan to add this" do you mean allowing `None` in addition to `-1` for SciPy?

Yes, for interoperability. I agree that `-1` is consistent with other places in `numpy` which was part of why NumPy's behavior was surprising to me. I think on their part it might have been less of a concious decision and more a result of the fact they call `fft(x, n=s[axis], axis)` internally.

**Comment by larsoner** [comment]
It might be worth opening a NumPy issue for clarification before working to implement the not-officially-supported method in SciPy. It's possible (even if a bit unlikely) they will decide to disable the behavior.

**Comment by lucascolley** [comment]
closed by numpy/numpy#25495. In line with the array API spec, `None` values in `s` have been deprecated in NumPy, and the `-1` behaviour now matches what is in SciPy.
